hscontrol/handlers.go

```diff
@@ -138,34 +141,39 @@ type registerWebAPITemplateConfig struct {
 	Key string
 }
 
+var codeStyleRegisterWebAPI = styles.Props{
+	styles.Display:         "block",
+	styles.Padding:         "20px",
+	styles.Border:          "1px solid #bbb",
+	styles.BackgroundColor: "#eee",
+}
+
+var htmlRegisterWebAPI = elem.Html(nil,
+	elem.Head(
+		nil,
+		elem.Title(
+			nil,
+			elem.Text("Registration - Headscale"),
+		),
+		elem.Meta(attrs.Props{
+			attrs.Name:    "viewport",
+			attrs.Content: "width=device-width, initial-scale=1",
+		},
+		),
+		elem.Body(attrs.Props{
+			attrs.Style: styles.Props{
+				styles.FontFamily: "sans",
+			}.ToInline(),
+		},
+			elem.H1(nil, elem.Text("headscale")),
+			elem.H2(nil, elem.Text("Machine registration")),
+			elem.P(nil, elem.Text("Run the command below in the headscale server to add this machine to your network:")),
+			elem.Code(attrs.Props{attrs.Style: codeStyleRegisterWebAPI.ToInline()},
+				elem.Raw(`headscale nodes register --user USERNAME --key {{.Key}}`)),
+		)))
+
 var registerWebAPITemplate = template.Must(
```

Comment:
then this can be removed and the places that calls it will call the function you create above with the render method.

Reply:
I made it a function that returns a *elem.Element type and the caller calls Render() on it. I [`defined the function](https://github.com/juanfont/headscale/pull/2120/commits/4d4964d1d13f224855f7c9eb62f2cddfed72b2ac#diff-b623bbee13a6f205d5f50b7d972d994547a961f40e25cbd5fd96478ce7325d96R149-R151). I didn't add error for the return I didn't find any place that needed it except if check for the config.Key is needed. Should I
[here](https://github.com/juanfont/headscale/pull/2120/commits/4d4964d1d13f224855f7c9eb62f2cddfed72b2ac#diff-b623bbee13a6f205d5f50b7d972d994547a961f40e25cbd5fd96478ce7325d96R209-R215) is the where the function is called